### PR TITLE
Add python testing

### PR DESCRIPTION
**Pull request by djmitche**
..Python testing for wihch all but one of the tests are marked "TODO", but enough to start writing a Python implementation.

### CODE REVIEW

File: specification.yml [hunk 1]
```diff
@@ -90,10 +90,6 @@ context:  {key: {}}
 template: 'hello ${key}'
 error:    true #TODO: Decide if it should be interpolated as JSON
 ---
-title: invalid context (1)
```
Comment (TheNavigat) [inline specification.yml]: Why is this deleted?

Comment (djmitche) [inline specification.yml]: It's invalid YAML, so it never makes it to the `jsone` call, which is what we're testing.  The JS YAML parser give an error parsing that particular section, but the Python parser gives an error for the entire file.

**Comment by djmitche** [comment]
I'll land it now.  This will get lots of adjustment later, anyway.
